### PR TITLE
HHH-11255  - NaturalId mapping fails when using a composite natural identifier

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-11255

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/cfg/IndexOrUniqueKeySecondPass.java [hunk 1]
```diff
@@ -64,10 +71,26 @@ public void doSecondPass(Map persistentClasses) throws MappingException {
...
+				Component component = Component.class.cast( property.getValue() );
+
+				List<Column> columns = new ArrayList<>();
+				component.getColumnIterator().forEachRemaining(selectable -> {
+					if(selectable instanceof Column) {
+						columns.add( (Column) selectable );
+					}
+				} );
+				addConstraintToColumns(columns);
+			} else {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/IndexOrUniqueKeySecondPass.java]: `else` on new line please

File: hibernate-core/src/main/java/org/hibernate/cfg/IndexOrUniqueKeySecondPass.java [hunk 2]
```diff
@@ -64,10 +71,26 @@ public void doSecondPass(Map persistentClasses) throws MappingException {
...
 			this.table = column.getTable();
-			addConstraintToColumn(
-					buildingContext.getMetadataCollector()
-							.getLogicalColumnName( table, column.getMappingColumn().getQuotedName() )
-			);
+
+			PersistentClass persistentClass = (PersistentClass) persistentClasses.get( column.getPropertyHolder().getEntityName() );
+			Property property = persistentClass.getProperty( column.getPropertyName() );
+
+			if(property.isComposite() && property.getValue() instanceof Component) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/IndexOrUniqueKeySecondPass.java]: Why even do the `#isComposite` check?  The instanceof is  enough

**Comment by vladmihalcea** [comment]
@sebersole I applied the proposed changes and I'm going to integrate it into master.

**Comment by sebersole** [comment]
Sounds good.  +1

**Pull request merged**